File: cpp/0111二叉树的最小深度.cpp
#include"mybitree.h"
#include<deque>

int minDepth(TreeNode *root)
{
    if(!root)
        return 0;
    int res = 0;
    deque<TreeNode *> nodes;
    nodes.push_back(root);

    while(!nodes.empty())
    {
        int size = nodes.size();
        ++res;
        for (int i = 0; i < size; ++i)
        {
            TreeNode *crt = nodes.front();
            nodes.pop_front();

            if(crt)
            {
                if(!crt->left&&!crt->right)
                    return res;
                nodes.push_back(crt->left);
                nodes.push_back(crt->right);
            }
        }
    }
    return res;
}

File: python/main/0111二叉树的最小深度.py
from binarytree import TreeNode

# 最小深度是指根节点到叶子节点的的最短距离
# 不要与求最大深度的算法混淆
def minDepth(root: TreeNode) -> int:
    if(not root):
        return 0
    if(root.left and not root.right):
        return 1+minDepth(root.left)
    if(root.right and not root.left):
        return 1+minDepth(root.right)
    
    return 1+min(minDepth(root.left),minDepth(root.right))